error_handling.py

```diff
@@ -0,0 +1,22 @@
+class CustomError(Exception):
+    def __init__(self, description: str, status_code: int, name: str):
+        self.description = description
+        self.status_code = status_code
+        self.name = name
+
+    def __str__(self):
+        return self.name
+
+
+class ValidationError(CustomError):
+    name = "Bad request"
+    status_code = 400
+
+    def __init__(self, description: str):
```

Comment:
Have you tested this part? Try to play with your API in Postman, make a request that should respond with `400` error

Reply:
And try a request that should raise `HTTPException` too

Reply:
Yes, I tested.

Reply:
![Снимок экрана (426)](https://user-images.githubusercontent.com/48601427/104589359-ca99ad80-567a-11eb-98d7-b04325067b5a.png)
![Снимок экрана (427)](https://user-images.githubusercontent.com/48601427/104589364-cc637100-567a-11eb-9896-86b0aab8d8ef.png)

Reply:
Results are similar

Reply:
well, got it. although it works, take a look:
```python
class CustomError(Exception):
    def __init__(self, description: str, status_code: int, name: str):
        self.description = description
        self.status_code = status_code
        self.name = name

    def __str__(self):
        return self.name
```
`description`, `status_code` and `name` are object fields, they are assigned to `self`.

At the same time:
```python
class WrongIdError(CustomError):
    name = "Not Found"
    status_code = 404

    def __init__(self, description: str):
        self.description = description

    def __str__(self):
        return self.name

Here the same fields are set as class-level, static members.

Because `CustomError` defines these as constructor arguments and `WrongIdError` inherits from it, it should be implemented as:
```python
class WrongIdError(CustomError):
    name = "Not Found"
    status_code = 404

    def __init__(self, description: str):
        super().__init__(name="Not Found", status_code=404, description=description)

    def __str__(self):
        return self.name
````

Quiz: why `handle_exception` works as expected?

I guess PyCharm should have emphasized that `WrongIdError.__init__` lacks a parent constructor call.

Both static member and instance members are ways to go, but the implementation should be consistent.
Here is how werkzeug implements its error classes:
https://github.com/pallets/werkzeug/blob/master/src/werkzeug/exceptions.py

Reply:
fixed